src/directedGraph.js

```diff
@@ -0,0 +1,103 @@
+import chalk from 'chalk'
+
+export default class DirectedGraph{
+  constructor(){
+    this.vertices = 0,
+    this.edges = [],
+    this.edgeCounter = 0
+  }
+  //inserts the vertices into the directed graph
+  insertVertices(vertices){
+    for( let i = 0; i < vertices.length; i++ ){
+      this.addVertex(vertices[i])
+      this.edges[vertices[i]].push(vertices[i])
+    }
+  }
+  //inserts the directions into the directed graph
+  insertDirections(directions){
+    for( let i = 0; i < directions.length; i++ ){
+      this.addDirection(directions[i].vertex1, directions[i].vertex2, directions[i].weight)
+    }
+  }
+  //adds a vertexical point on a directed graph
+  addVertex(vertex){
+    this.edges[vertex] = []
+    this.vertices++
+  }
+
+  //checks if a vertex exists on the directed graph
+  hasVertex(vertex){
+    for(let i = 0; i < this.vertices; i++){
+      if(this.edges[i] === vertex){
+        return true
+      }
+    }
+    return false
+  }
+  //adds a vector connecting given vertices with a weight attached
+  addDirection(vertex1, vertex2, weight){
+    // console.log(this.edges[vertex1])
+    this.edges[vertex1].push({vertex1:vertex1, vertex2:vertex2, weight:weight})
+  }
+  //checks the directed graph if it has a particular direction
+  hasDirection(vertex1, vertex2, dir){
```

Comment:
try refactoring this function, there are a lot of nested ```if/else```. See if it is parts of this function can be moved out into their own functions. There is the Single Responsibility Principle https://en.wikipedia.org/wiki/Single_responsibility_principle